### PR TITLE
Feat: manage biobank PIDs

### PR DESCRIPTION
**Pull request by tommydeboer**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/system tested
- [x] User documentation updated
- [x] Clean commits
- [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by tommydeboer** [comment]
[[PID] Mechanism to pass handle credentials to publishing script](https://trello.com/c/LIIyI32c/478-pid-mechanism-to-pass-handle-credentials-to-publishing-script)

**Comment by tommydeboer** [comment]
[[PID] Assign PIDs to existing biobanks](https://trello.com/c/71jeJOVZ/483-pid-assign-pids-to-existing-biobanks)

File: scripts/biobank_pid_assignment.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  "certificate_only": "<FULL_PATH_TO_CERTIFICATE_ONLY.pem>",
+  "client": "rest",
+  "prefix": "<HANDLE_PREFIX>"
+}
+
+See https://eudat-b2handle.github.io/PYHANDLE/pyhandleclientrest.html#authentication
+-using-client-certificates for more information.
+
+Follow these steps:
+1. Add an attribute "pid" to the table
```
Comment (jelmerveen) [inline scripts/biobank_pid_assignment.py:26]: what type should it be?

File: scripts/biobank_pid_assignment.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+2. Make the attribute "nullable"
+3. Run the script
```
Comment (jelmerveen) [inline scripts/biobank_pid_assignment.py:28]: Run the script where?

File: scripts/biobank_pid_assignment.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+4. Make the "pid" attribute "required" (not nullable) and "readOnly"
```
Comment (jelmerveen) [inline scripts/biobank_pid_assignment.py:29]: Change the "pid" from nullable to

File: scripts/biobank_pid_assignment.py (hunk 1 above)
Comment (jelmerveen) [reply scripts/biobank_pid_assignment.py:26]: and what table

File: scripts/biobank_pid_assignment.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+from dotenv import dotenv_values
+from pyhandle.clientcredentials import PIDClientCredentials
+from pyhandle.handleclient import PyHandleClient
+
+from molgenis.bbmri_eric.bbmri_client import ExtendedSession
+
+table = "eu_bbmri_eric_biobanks"
+
+print("Logging in to the directory")
```
Comment (jelmerveen) [inline scripts/biobank_pid_assignment.py:40]: the bbmri-app is called the directory, so this must be something else?

File: scripts/biobank_pid_assignment.py (hunk 2 above)
Comment (tommydeboer) [reply scripts/biobank_pid_assignment.py:28]: Anywhere :)

File: scripts/biobank_pid_assignment.py (hunk 4 above)
Comment (tommydeboer) [reply scripts/biobank_pid_assignment.py:40]: No, it _is_ the directory :)

**Comment by tommydeboer** [comment]
[[PID] Assign PIDs to new biobanks](https://trello.com/c/yypVvjVB/486-pid-assign-pids-to-new-biobanks)

**Comment by tommydeboer** [comment]
[[PID] Update handle on biobank name change](https://trello.com/c/MWUVVny4/488-pid-update-handle-on-biobank-name-change)

**Comment by tommydeboer** [comment]
[[PID] Update STATUS of deleted biobanks](https://trello.com/c/lSmrz81v/497-pid-update-status-of-deleted-biobanks)

**Comment by tommydeboer** [comment]
[[PID] Write unit tests](https://trello.com/c/tpqlh2qz/500-pid-write-unit-tests)

File: scripts/biobank_pid_assignment.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+session.login(username, password)
+
+print("Creating handle client")
+pid_service = PidService.from_credentials("pyhandle_creds.json")
+
+print("Getting data from the directory")
+biobanks = session.get_uploadable_data(table)
+
+print("Registering PIDs")
+url_prefix = session.url.rstrip("/") + "/#/biobank/"
```
Comment (jelmerveen) [inline scripts/biobank_pid_assignment.py:54]: why stripping the / first and then adding it?

wouldn't this be sufficient? 

```
session.url + "#/biobank/"
```

File: src/molgenis/bbmri_eric/pid_manager.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+class PidManager:
+    """
+    This class is responsible for managing PIDs of BBMRI-ERIC entities: assignment,
+    updates en status changes are done here.
+    """
+
+    def __init__(self, pid_service: PidService, printer: Printer, url: str):
+        self.pid_service = pid_service
+        self.printer = printer
+        self.biobank_url_prefix = url.rstrip("/") + "/#/biobank/"
```
Comment (jelmerveen) [inline src/molgenis/bbmri_eric/pid_manager.py:18]: same here?

File: tests/test_printer.py [hunk 7]
```diff
@@ -22,7 +23,8 @@ def test_indentation(capsys):
     printer.indent()
     printer.print("line3")
     printer.dedent()
-    printer.print("line4")
+    printer.print("line4", 1)
+    printer.print("line5")
 
     captured = capsys.readouterr()
```
Comment (jelmerveen) [inline tests/test_printer.py:29]: its offtopic and out of scope but would suggest to change this to readout_err for better readability

**Review by jelmerveen — APPROVED** [review]

**Pull request merged**